File: app/domain/security/pqc/kyber/ffi/ios/AfetNetKyberBridge.swift
// @afetnet: iOS Swift bridge for Kyber1024 post-quantum key exchange
// Provides safe interface to Rust Kyber implementation

import Foundation

@objc(AfetNetKyberBridge)
public class AfetNetKyberBridge: NSObject {

    // @afetnet: Generate Kyber1024 keypair
    @objc static func generateKeypair() -> String? {
        // Call Rust function via bridging header
        let resultPtr = kyber_generate_keypair()

        guard let resultPtr = resultPtr else {
            NSLog("Failed to generate Kyber keypair")
            return nil
        }

        let result = String(cString: resultPtr)
        free_string(resultPtr)

        return result
    }

    // @afetnet: Encapsulate shared secret
    @objc static func encapsulate(publicKeyHex: String) -> String? {
        let publicKeyCStr = publicKeyHex.cString(using: .utf8)

        guard let publicKeyCStr = publicKeyCStr else {
            NSLog("Invalid public key string")
            return nil
        }

        let resultPtr = kyber_encapsulate(publicKeyCStr)

        guard let resultPtr = resultPtr else {
            NSLog("Failed to encapsulate")
            return nil
        }

        let result = String(cString: resultPtr)
        free_string(resultPtr)

        return result
    }

    // @afetnet: Decapsulate shared secret
    @objc static func decapsulate(secretKeyHex: String, ciphertextHex: String) -> String? {
        let secretKeyCStr = secretKeyHex.cString(using: .utf8)
        let ciphertextCStr = ciphertextHex.cString(using: .utf8)

        guard let secretKeyCStr = secretKeyCStr, let ciphertextCStr = ciphertextCStr else {
            NSLog("Invalid input strings")
            return nil
        }

        let resultPtr = kyber_decapsulate(secretKeyCStr, ciphertextCStr)

        guard let resultPtr = resultPtr else {
            NSLog("Failed to decapsulate")
            return nil
        }

        let result = String(cString: resultPtr)
        free_string(resultPtr)

        return result
    }
}

File: app/domain/security/pqc/kyber/ffi/android/AfetNetKyberBridge.kt
// @afetnet: Android Kotlin bridge for Kyber1024 post-quantum key exchange
// Provides safe interface to Rust Kyber implementation

package com.afetnet.bridge.kyber

object AfetNetKyberBridge {

    // @afetnet: Load native library
    init {
        System.loadLibrary("afetnet_kyber")
    }

    // @afetnet: Generate Kyber1024 keypair
    @JvmStatic
    external fun generateKeypair(): String?

    // @afetnet: Encapsulate shared secret
    @JvmStatic
    external fun encapsulate(publicKeyHex: String): String?

    // @afetnet: Decapsulate shared secret
    @JvmStatic
    external fun decapsulate(secretKeyHex: String, ciphertextHex: String): String?

    // @afetnet: Free native string
    @JvmStatic
    external fun freeString(s: String?)
}


























